scene/resources/particle_process_material.cpp

```diff
@@ -830,6 +830,13 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "	TRANSFORM[3].z = 0.0;\n";
 	}
 
+	// scale by scale
+	code += "	float base_scale = mix(scale_min, scale_max, scale_rand);\n";
+	code += "	base_scale = sign(base_scale) * max(abs(base_scale), 0.001);\n";
+	code += "	TRANSFORM[0].xyz *= base_scale * sign(tex_scale.r) * max(abs(tex_scale.r), 0.001);\n";
+	code += "	TRANSFORM[1].xyz *= base_scale * sign(tex_scale.g) * max(abs(tex_scale.g), 0.001);\n";
+	code += "	TRANSFORM[2].xyz *= base_scale * sign(tex_scale.b) * max(abs(tex_scale.b), 0.001);\n";
+
```

Comment:
It looks like this was originally above the ``if (COLLIDED)`` branch but was moved to be below in https://github.com/godotengine/godot/pull/55387. 

What is the reasoning for bringing it back up? In either case it looks like it shouldn't interact with the ``if (COLLIDED)`` branch